### PR TITLE
format chart number values

### PR DESCRIPTION
**Pull request by mariamiah**
- Group numbers with commas
- On axes, represent numbers in short formats e.g 10K, 2M e.t.c

#95

### CODE REVIEW

File: components/ComparisonChartDataHandler/ComparisonChartDataHandler.tsx [hunk 1]
```diff
@@ -109,6 +109,18 @@ const ComparisonChartDataHandler: FunctionComponent<ComponentProps> = ({ data, l
...
+                    dataFormat: props.indicators[0].data_format,
+                    prefix: props.indicators[0].value_prefix,
+                    suffix: props.indicators[0].value_suffix
+                  },
+                  {
+                    dataFormat: props.indicators[1].data_format,
+                    prefix: props.indicators[1].value_prefix,
+                    suffix: props.indicators[1].value_suffix
+                  }
+                ]}
```
Comment (edwinmp) [inline components/ComparisonChartDataHandler/ComparisonChartDataHandler.tsx:123]: Rewrite this with less lines ... use `props.indicators.map`

Comment (mariamiah) [reply components/ComparisonChartDataHandler/ComparisonChartDataHandler.tsx:123]: okay

File: components/LocationComparisonBarChart/LocationComparisonBarChart.tsx [hunk 2]
```diff
@@ -23,18 +25,45 @@ const LocationComparisonBarChart: FunctionComponent<LocationComparisonChartProps
...
+          return `<div>${name} <ul><li>${seriesName}: ${addPrefixAndSuffix(
+            formatNumber(value, 1),
+            valueOptions[1]
+          )}</li></ul></div>`;
+        }
+
+        return `<div>${name} <ul><li>${seriesName}: ${addPrefixAndSuffix(
+          formatNumber(value, 1),
+          valueOptions[0]
+        )}</li></ul></div>`;
```
Comment (edwinmp) [inline components/LocationComparisonBarChart/LocationComparisonBarChart.tsx:44]: There's repetition here. Extract into a function and reuse that.

File: components/LocationComparisonBarChart/LocationComparisonBarChart.tsx [hunk 3]
```diff
@@ -23,18 +25,45 @@ const LocationComparisonBarChart: FunctionComponent<LocationComparisonChartProps
...
       }
     },
     xAxis: [
       {
         type: 'value',
-        position: 'top'
+        position: 'top',
+        axisLabel: {
+          formatter: (value: number): string => {
+            return addPrefixAndSuffix(formatNumber(value, 0), valueOptions[0]);
```
Comment (edwinmp) [inline components/LocationComparisonBarChart/LocationComparisonBarChart.tsx:53]: Could you please check for if the value is 0 and exclude that from the formatting... That way, you'll not have the strange appearance below.

image

File: components/LocationComparisonBarChart/LocationComparisonBarChart.tsx [hunk 4]
```diff
@@ -23,18 +25,45 @@ const LocationComparisonBarChart: FunctionComponent<LocationComparisonChartProps
...
       },
       {
         type: 'value',
         gridIndex: 1,
         position: 'top',
-        inverse: true
+        inverse: true,
+        axisLabel: {
+          formatter: (value: number): string => {
+            return addPrefixAndSuffix(formatNumber(value, 0), valueOptions[1]);
```
Comment (edwinmp) [inline components/LocationComparisonBarChart/LocationComparisonBarChart.tsx:64]: Same here.

File: components/LocationComparisonBarChart/LocationComparisonBarChart.tsx [hunk 5]
```diff
@@ -24,18 +26,39 @@ const LocationComparisonBarChart: FunctionComponent<LocationComparisonChartProps
...
+
+      formatter: (params: EChartOption.Tooltip.Format[]): string => {
+        const { seriesName, name, seriesIndex } = params[0];
+        const { value } = params[0] as { value: number };
+
+        if (seriesIndex === 1) {
+          return formatSeries(1, name, seriesName, value, valueOptions);
+        }
+
+        return formatSeries(0, name, seriesName, value, valueOptions);
```
Comment (edwinmp) [inline components/LocationComparisonBarChart/LocationComparisonBarChart.tsx:39]: Apply previous comments here as well.

File: components/LocationComparisonBarChart/LocationComparisonBarChart.tsx [hunk 6]
```diff
@@ -24,18 +26,39 @@ const LocationComparisonBarChart: FunctionComponent<LocationComparisonChartProps
...
     },
     xAxis: [
       {
         type: 'value',
-        position: 'top'
+        position: 'top',
+        axisLabel: {
+          formatter: (value: number): string => {
+            return value === 0 ? '0' : addPrefixAndSuffix(formatNumber(value, 0), valueOptions[0]);
+          }
```
Comment (edwinmp) [inline components/LocationComparisonBarChart/LocationComparisonBarChart.tsx:49]: Remove return & curly braces ...

File: components/LocationComparisonBarChart/LocationComparisonBarChart.tsx [hunk 7]
```diff
@@ -24,18 +26,39 @@ const LocationComparisonBarChart: FunctionComponent<LocationComparisonChartProps
...
       },
       {
         type: 'value',
         gridIndex: 1,
         position: 'top',
-        inverse: true
+        inverse: true,
+        axisLabel: {
+          formatter: (value: number): string => {
+            return value === 0 ? '0' : addPrefixAndSuffix(formatNumber(value, 0), valueOptions[1]);
```
Comment (edwinmp) [inline components/LocationComparisonBarChart/LocationComparisonBarChart.tsx:59]: same here

File: utils/values.ts [hunk 8]
```diff
@@ -15,3 +15,15 @@ export const DEFAULT_VALUE = 'No Data';
...
+  name: string | undefined,
+  seriesName: string | undefined,
+  value: number,
+  valueOptions: ValueOptions[]
+): string => {
+  return `<div>${name}<ul><li>${seriesName}: ${addPrefixAndSuffix(
+    formatNumber(value, 1),
+    valueOptions[index]
+  )}</li></ul></div>`;
+};
```
Comment (edwinmp) [inline utils/values.ts:29]: Rewrite this function without passing in the index.

**Pull request merged**